merkle-tree/src/routes/merkle-tree.routes.js

```diff
@@ -21,48 +22,80 @@ const alreadyStarting = {}; // initialises as false
 async function startEventFilter(req, res, next) {
   logger.debug('src/routes/merkle-tree.routes startEventFilter()');
 
-  const { contractName, treeId, contractAddress } = req.body; // contractAddress & treeId are optional parameters. Address can instead be inferred by Timber in many cases.
+  const { contractName, treeId, contractId } = req.body; // contractAddress & treeId are optional parameters. Address can instead be inferred by Timber in many cases.
   const { db } = req.user;
 
+  logger.debug(
+    `Received data: contractName: ${contractName}, treeId: ${treeId}, contractId: ${contractId}`,
+  );
   // TODO: if possible, make this easier to read and follow. Fewer 'if' statements. Perhaps use 'switch' statements instead?
   try {
-    if (alreadyStarted[contractName] && (treeId === undefined || treeId === '')) {
-      res.data = { message: `filter already started for ${contractName}` };
-    } else if (alreadyStarted[(contractName, treeId)]) {
-      res.data = { message: `filter already started for ${contractName}.${treeId}` };
-    } else if (alreadyStarting[contractName] && (treeId === undefined || treeId === '')) {
+    if (
+      alreadyStarted[`${contractName} ${contractId}`] &&
```

Comment:
In the case of no `contractID` we would be storing the started info under `name ` (with a space). I'm not sure but this may turn out to be `alreadyStarted[MyContract undefined]` which is not ideal. Could we instead do something like a tuple e.g. `alreadyStarted[(contractName, contractId, treeId)]` or a string:
```
alreadyStarted[`${contractName}${contractId ? '_' + contractId : ''}${treeId ? '_' + treeId : ''}`]
```

Reply:
Good idea, noted

Reply:
Changed to the suggested code.